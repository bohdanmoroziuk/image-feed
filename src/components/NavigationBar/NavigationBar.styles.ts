import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    height: 40,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'rgba(0, 0, 0, 0.1)',
    alignItems: 'center',
    justifyContent: 'center',
  },
  closeButton: {
    position: 'absolute',
    left: 20,
    top: 0,
    bottom: 0,
    justifyContent: 'center'
  },
  title: {
    fontWeight: '500',
  },
});
