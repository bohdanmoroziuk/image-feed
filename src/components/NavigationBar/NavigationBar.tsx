import { FunctionComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

import styles from './NavigationBar.styles';

export interface NavigationBarProps {
  title: string;
  closeButtonText: string;
  onClose: () => void;
}

export const NavigationBar: FunctionComponent<NavigationBarProps> = ({
  title,
  closeButtonText,
  onClose,
}) => (
  <View style={styles.container}>
    <TouchableOpacity
      style={styles.closeButton}
      onPress={onClose}
    >
      <Text>
        {closeButtonText}
      </Text>
    </TouchableOpacity>
    <Text style={styles.title}>
      {title}
    </Text>
  </View>
);
