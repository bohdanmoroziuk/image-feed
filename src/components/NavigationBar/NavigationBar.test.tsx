import { render, fireEvent } from '@testing-library/react-native';

import { NavigationBar } from './NavigationBar';

afterEach(() => {
  jest.clearAllMocks();
});

const mockOnClose = jest.fn();

describe('NavigationBar', () => {
  it('renders successfully', () => {
    const { toJSON } = render(
      <NavigationBar
        title="Comments"
        closeButtonText="Close"
        onClose={mockOnClose}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('calls the onClose handler when the close button is pressed', () => {
    const { getByText } = render(
      <NavigationBar
        title="Comments"
        closeButtonText="Close"
        onClose={mockOnClose}
      />
    );

    fireEvent.press(getByText('Close'));

    expect(mockOnClose).toHaveBeenCalled();
  });
});
