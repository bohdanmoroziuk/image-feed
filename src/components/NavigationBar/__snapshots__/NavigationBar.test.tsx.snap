// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "rgba(0, 0, 0, 0.1)",
      "borderBottomWidth": 0.5,
      "height": 40,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "justifyContent": "center",
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Text>
      Close
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontWeight": "500",
      }
    }
  >
    Comments
  </Text>
</View>
`;
